base/src/main/java/org/aya/core/visitor/MonoidalVarFolder.java

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.core.visitor;
+
+import kala.collection.SeqLike;
+import kala.collection.SeqView;
+import org.aya.core.Matching;
+import org.aya.core.def.*;
+import org.aya.core.term.*;
+import org.aya.generic.Arg;
+import org.aya.ref.DefVar;
+import org.aya.ref.Var;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Function;
+
+public interface MonoidalVarFolder<R> extends Function<Term, R> {
+  // TODO: Do we need to visit variables in `access` and `new` as well?
```

Comment:
I think so.

Reply:
This comment is actually from the old version of the code.
Do I need to change in this PR anyway?

Reply:
Well let's leave it then. I think I'll handle it (or you can, if you are fast enough ;p)